File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/receipt/ReceiptAddressRequest.java
package com.novax.sdk.pay.receipt;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import com.novax.sdk.pay.model.PayOrderAddressFixedResp;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.Map;

/** GET /pay/v3/receipt/address — fixed receipt address lookup. */
@Getter
@Builder
public class ReceiptAddressRequest extends AbstractApiRequest<PayOrderAddressFixedResp> {

    private final String protocol;
    private final String smartContractAddress;
    private final String companyUserId;

    @Override public HttpMethod method() { return HttpMethod.GET; }
    @Override public String path() { return "/pay/v3/receipt/address"; }

    @Override public Map<String, ?> queryParams() {
        Map<String, Object> q = new LinkedHashMap<>();
        if (protocol != null) q.put("protocol", protocol);
        if (smartContractAddress != null) q.put("smartContractAddress", smartContractAddress);
        if (companyUserId != null) q.put("companyUserId", companyUserId);
        return q;
    }

    @Override public TypeRef<PayOrderAddressFixedResp> responseType() {
        return TypeRef.of(PayOrderAddressFixedResp.class);
    }
}

File: python/novax_sdk/pay/receipt/receipt_address_request.py
from dataclasses import dataclass
from typing import Any

from novax_sdk.pay.model import PayOrderAddressFixedResp
from novax_sdk.request.api_request import ApiRequest, HttpMethod


@dataclass
class ReceiptAddressRequest(ApiRequest):
    """GET /pay/v3/receipt/address"""

    protocol: str | None = None
    smart_contract_address: str | None = None
    company_user_id: str | None = None

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.GET

    @property
    def path(self) -> str:
        return "/pay/v3/receipt/address"

    def query_params(self) -> dict[str, Any]:
        return {k: v for k, v in {
            "protocol": self.protocol,
            "smartContractAddress": self.smart_contract_address,
            "companyUserId": self.company_user_id,
        }.items() if v is not None}

    @property
    def response_type(self) -> type:
        return PayOrderAddressFixedResp
